lua/neogit/popups/branch/actions.lua

```diff
@@ -22,6 +22,40 @@ local function parse_remote_branch_name(ref)
   return remote, branch_name
 end
 
+M.spin_off_branch = operation("spin_off_branch", function()
+  if #git.repo.staged.items > 0 or #git.repo.unstaged.items > 0 then
```

Comment:
If we do a quick stash here, before the reset, can that just be restored once checking out the new branch?